controllers/datadogagent/feature/logcollection/feature.go

```diff
@@ -0,0 +1,171 @@
+		f.enable = true
+		if logCollection.ContainerCollectAll != nil && apiutils.BoolValue(logCollection.ContainerCollectAll) {
+			f.containerCollectAll = true
+		}
+		if apiutils.BoolValue(logCollection.ContainerCollectUsingFiles) {
+			f.containerCollectUsingFiles = true
+		}
+		f.containerLogsPath = *logCollection.ContainerLogsPath
+		f.podLogsPath = *logCollection.PodLogsPath
+		f.containerSymlinksPath = *logCollection.ContainerSymlinksPath
+		f.tempStoragePath = *logCollection.TempStoragePath
+		if logCollection.OpenFilesLimit != nil {
+			f.openFilesLimit = *logCollection.OpenFilesLimit
+		}
+
+		reqComp = feature.RequiredComponents{
+			Agent: feature.RequiredComponent{
+				IsRequired: &f.enable,
+				Containers: []apicommonv1.AgentContainerName{
+					apicommonv1.CoreAgentContainerName,
+				},
+			},
+		}
+	}
+	return reqComp
+}
+
+// ConfigureV1 use to configure the feature from a v1alpha1.DatadogAgent instance.
+func (f *logCollectionFeature) ConfigureV1(dda *v1alpha1.DatadogAgent) (reqComp feature.RequiredComponents) {
+	logCollection := dda.Spec.Features.LogCollection
+
+	if logCollection != nil && apiutils.BoolValue(logCollection.Enabled) {
+		f.enable = true
+		if apiutils.BoolValue(logCollection.LogsConfigContainerCollectAll) {
+			f.containerCollectAll = true
+		}
+		if apiutils.BoolValue(logCollection.ContainerCollectUsingFiles) {
+			f.containerCollectUsingFiles = true
+		}
+		f.containerLogsPath = *logCollection.ContainerLogsPath
+		f.podLogsPath = *logCollection.PodLogsPath
+		f.containerSymlinksPath = *logCollection.ContainerSymlinksPath
+		f.tempStoragePath = *logCollection.TempStoragePath
+		if logCollection.OpenFilesLimit != nil {
+			f.openFilesLimit = *logCollection.OpenFilesLimit
+		}
+
+		reqComp = feature.RequiredComponents{
+			Agent: feature.RequiredComponent{
+				IsRequired: &f.enable,
+				Containers: []apicommonv1.AgentContainerName{
+					apicommonv1.CoreAgentContainerName,
+				},
+			},
+		}
+	}
+	return reqComp
+}
+
+// ManageDependencies allows a feature to manage its dependencies.
+// Feature's dependencies should be added in the store.
+func (f *logCollectionFeature) ManageDependencies(managers feature.ResourceManagers) error {
+	return nil
+}
+
+// ManageClusterAgent allows a feature to configure the ClusterAgent's corev1.PodTemplateSpec
+// It should do nothing if the feature doesn't need to configure it.
+func (f *logCollectionFeature) ManageClusterAgent(managers feature.PodTemplateManagers) error {
+	return nil
+}
+
+// ManageNodeAgent allows a feature to configure the Node Agent's corev1.PodTemplateSpec
+// It should do nothing if the feature doesn't need to configure it.
+func (f *logCollectionFeature) ManageNodeAgent(managers feature.PodTemplateManagers) error {
+	if f.enable {
+		// pointerdir volume mount
+		pointerVol, pointerVolMount := volume.GetVolumes(apicommon.PointerVolumeName, f.tempStoragePath, apicommon.PointerVolumePath, apicommon.PointerVolumeReadOnly)
```

Comment:
i wonder whether it's worth having a `readOnly` const for each mount.. it may become cumbersome for something that isn't likely to change or particularly useful to track (open to debate). as an alternative, maybe we can use generic boolean constants such as 
```
const trueValue = true
const falseValue = false
```

Reply:
Discussed separately. Decided on using `true`/`false` instead of adding more constants.